litestar/contrib/opentelemetry/plugin.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from litestar.contrib.opentelemetry.config import OpenTelemetryConfig
+from litestar.contrib.opentelemetry.middleware import OpenTelemetryInstrumentationMiddleware
+from litestar.middleware.base import DefineMiddleware
+from litestar.plugins import InitPluginProtocol
+
+if TYPE_CHECKING:
+    from litestar.config.app import AppConfig
+    from litestar.types.composite_types import Middleware
+
+
+class OpenTelemetryPlugin(InitPluginProtocol):
+    """OpenTelemetry Plugin."""
+
+    __slots__ = ("config", "_middleware")
+
+    def __init__(self, config: OpenTelemetryConfig | None = None) -> None:
+        self.config = config or OpenTelemetryConfig()
+        self._middleware: DefineMiddleware | None = None
+        super().__init__()
+
+    @property
+    def middleware(self) -> DefineMiddleware:
+        if self._middleware:
+            return self._middleware
+        return DefineMiddleware(OpenTelemetryInstrumentationMiddleware, config=self.config)
+
+    def on_app_init(self, app_config: AppConfig) -> AppConfig:
+        app_config.middleware, _middleware = self.pop_otel_middleware(app_config.middleware)
+        return app_config
+
+    @staticmethod
+    def pop_otel_middleware(middlewares: list[Middleware]) -> tuple[list[Middleware], DefineMiddleware | None]:
```

Comment:
Should we make this method private in case we come up with a better way to handle this in the future?

Reply:
Yeah, probably. Not great calling into a private method from inside the app object either, but I guess it's better than having this in the open.